robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -288,7 +288,7 @@ def __init__(self, robot):
     def execute(self, userdata):
         """ Execute hook """
         if hasattr(userdata, 'arm'):
-            arm = self._robot.get_arm(userdata.arm)[0]  # Using userdata makes sure we don't need more arm designator magic
+            arm = self._robot.get_robot_arm(gripper_type="yes", preferred=userdata.arm)
```

Comment:
Here we do require a very specific arm. Otherwise, one state part of Grab uses a different arm than the other.